lib/internal/modules/esm/hooks.js

```diff
@@ -114,6 +114,25 @@ class Hooks {
     this.addCustomLoaders(userLoaders);
   }
 
+  /**
+   * Import and register custom/user-defined module loader hook(s).
+   * @param {string} urlOrSpecifier
+   */
+  async registerLoader(urlOrSpecifier) {
+    const { DefaultModuleLoader } = require('internal/modules/esm/loader')
+
+    const moduleLoader = new DefaultModuleLoader();
+    const parentURL = pathToFileURL(process.cwd() + '/').href;
```

Comment:
`process.cwd()` can throw after user code starts running: https://github.com/nodejs/node/blob/b31d587dc8c1b0f459243d2b96a17a7175c996cf/lib/internal/process/esm_loader.js#L22-L28

Reply:
I could create a simple function called `safeProcessCwd()` and use it in `esm_loader.js` and also in `hooks.js`. What do you think?

Is there a correct place where I can create this function? I'm thinking about `internal/util`.